fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/ConnectingActivity.java

```diff
@@ -0,0 +1,117 @@
+package com.github.dedis.popstellar.ui.home;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.model.objects.Lao;
+import com.github.dedis.popstellar.repository.remote.GlobalNetworkManager;
+import com.github.dedis.popstellar.ui.detail.LaoDetailActivity;
+import com.github.dedis.popstellar.utility.Constants;
+import com.github.dedis.popstellar.utility.error.ErrorUtils;
+import com.tinder.scarlet.WebSocket;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.observables.ConnectableObservable;
+
+@AndroidEntryPoint
+public class ConnectingActivity extends AppCompatActivity {
+  public static final String TAG = ConnectingActivity.class.getSimpleName();
+
+  private final CompositeDisposable disposables = new CompositeDisposable();
+
+  @Inject GlobalNetworkManager networkManager;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.connecting_activity);
+    String channelId = (String) getIntent().getExtras().get(Constants.LAO_ID_EXTRA);
+    setupCancelButton();
```

Comment:
Is it standard practice to put private functions such as this one after protected ones ? 
It seems a bit unnatural for me to have to scroll down 50 lines to discover the 2 lines of code this call hides :sweat_smile:

Reply:
I don't know about any convention regarding the matter. But in the project, in activities this is consistent (at least mostly) as the onCreate is the first function usually. 
I could have organized this a tad better for github reviews though I will admit :). For IDE browsing it's not a problem IMO since ctrl click does wonders.
I'll rearrange the private methods a bit